lang/java/reef-io/src/main/java/org/apache/reef/io/data/loading/api/DataLoadingRequestBuilder.java

```diff
@@ -258,14 +303,25 @@ public Configuration build() throws BindException {
       }
     }
 
-    return jcb
-        .bindNamedParameter(LoadDataIntoMemory.class, Boolean.toString(this.inMemory))
-        .bindConstructor(InputFormat.class, InputFormatExternalConstructor.class)
-        .bindConstructor(JobConf.class, JobConfExternalConstructor.class)
-        .bindNamedParameter(JobConfExternalConstructor.InputFormatClass.class, inputFormatClass)
-        .bindNamedParameter(JobConfExternalConstructor.InputPath.class, inputPath)
-        .bindImplementation(DataLoadingService.class, InputFormatLoadingService.class)
-        .build();
+    jcb.bindNamedParameter(LoadDataIntoMemory.class, Boolean.toString(this.inMemory))
+       .bindNamedParameter(JobConfExternalConstructor.InputFormatClass.class, inputFormatClass);
+
+
+    for (final DataPartition partition : partitions) {
+      jcb.bindSetEntry(LocationAwareJobConfs.DataPartitions.class, DataPartitionSerializer.serialize(partition));
+    }
+
+    // we do this check for backwards compatibility, if there's a single partition, we just use the
```

Comment:
It might be easier to just have a boolean flag indicating that someone used the backwards-compat APIs. This `if` condition could fire even for users of the new API.